unicodetools/data/linkification/dev/LinkDetectionTest.txt

```diff
@@ -0,0 +1,432 @@
+See example.😎  on…
+See .fr on…
+
+# Sample uppercase
+See ⸠TEST.COM⸡ on…
+See ⸠FOO.VERMÖGEN.com⸡ on…
+
+# Illegal domain names, so not linked
+See http://.foo.example.com/αβγ on…
+See http://foo..example.com/αβγ on…
+See http://-foo.example-.com. on…
+
+# Legal but unusual. Because we might be at the end of a sentence, we don't include the . unless followed by a Path, Query, or Fragment
+See ⸠http://foo.example.com⸡. on…
+See ⸠http://foo.example.com./αβγ⸡ on…
+
+# Various Path/Query/Fragment cases
+See ⸠example.com⸡! on…
+See ⸠example.com/αβγ⸡ on…
+See ⸠example.com?αβγ⸡ on…
+See ⸠example.com#αβγ⸡ on…
+See ⸠example.com/αβγ/δεζ?θικ#λμν⸡ on…
+See ⸠example.com/αβγ/δεζ?δ.εφ#λμν⸡ on…
+See ⸠https://example.com/αβγ⸡ on…
+See ⸠http://example.com/αβγ⸡ on…
+
+# Illustrations of Soft vs Hard, bracketing
+
+# Break on hard (such as ' ')
+See ⸠example.com/αβγ/δεζ?δ⸡ εφ#λμν on…
+
+# Break on soft (such as '.') followed by hard (such as ' ')
+See ⸠example.com/αβγ/δεζ?δ⸡. εφ#λμν on…
+See ⸠example.com/α/βγ?δ/ε?ζ#λ/μ?ν#π⸡ on…
+
+# Break on unmatched bracket
+See ⸠example.com/αβ⸡) on…
+
+# Include matched bracket
+See ⸠example.com/α(β)⸡ on…
+See ⸠example.com/αβγ/δρς?α=θ&β=κ#λμν:~:text=φχψ⸡ on…
+
+# Continue past matching brackets within same part and not across interior syntax
+See ⸠example.com/α(β)γ/δρς?α=(θ)&(β)=κ#λμν:~:text=(φχψ)⸡ on…
+See ⸠example.com/α[(β)]γ/δρς?α=[(θ)]&[(β)]=κ#λμν:~:text=[(φχψ)]⸡ on…
+
+# Don't match mismatched brackets
+See ⸠example.com/α[(β⸡])γ/δρς?α=[(θ)]&[(β)]=κ#λμν:~:text=φχψ on…
+See ⸠example.com/α[(β)]γ/δρς?α=[(θ⸡])&[(β)]=κ#λμν:~:text=φχψ on…
+See ⸠example.com/α[(β)]γ/δρς?α=[(θ)]&[(β⸡])=κ#λμν:~:text=φχψ on…
+See ⸠example.com/α[(β)]γ/δρς?α=[(θ)]&[(β)]=κ#λμν:~:text=[(φχψ⸡]) on…
+
+# Don't match across interior syntax 
+See ⸠example.com/αβ(γ/δ⸡)ρς?α=θ&β=κ#λμν:~:text=φχψ on…
+See ⸠example.com/αβγ/δρς?α(=⸡)θ&β=κ#λμν:~:text=φχψ on…
+See ⸠example.com/αβγ/δρς?α=θ(&⸡)β=κ#λμν:~:text=φχψ on…
+See ⸠example.com/αβγ/δρς?α=θ&β=κ#λμ(ν:~:text=φχ⸡)ψ on…
+
+⸠https://ja.wikipedia.org/wiki/フィンセント・ファン・ゴッホ⸡
+⸠https://ja.wikipedia.org/wiki/%E3%83%95%E3%82%A3%E3%83%B3%E3%82%BB%E3%83%B3%E3%83%88%E3%83%BB%E3%83%95%E3%82%A1%E3%83%B3%E3%83%BB%E3%82%B4%E3%83%83%E3%83%9B⸡
+⸠abc.com⸡ ⸠deΩ.uk⸡ ⸠foo.accountants⸡ ⸠foo.香港⸡
+⸠abc.com/d/e?xyz#w⸡
+⸠abc.XN--11B4C3D⸡
+
+# Email addresses
+
+See ⸠αβγ.δεζ@example.com⸡ on…
+See ⸠mailto:αβγ.δεζ@example.com⸡ on…
+
+# If a local-part is invalid, skip the domain name
+See mailto:αβγ.δεζ.@example.com on…
+See mailto:αβγ..δεζ@example.com on…
+See mailto:.αβγ.δεζ@example.com on…
+
+#Stop backing up when a space is hit
+See ⸠abcd@example.com⸡
+
+#Include the medial dot
+See ⸠x.abcd@example.com⸡.
+
+#Handle non-ASCII
+See ⸠アルベルト.アルベルト@example.com⸡
+ 
+#Invalid domain name, so not linked
+See @example.😎
+
+# No valid local-part, so not linked
+See @example.com
+See john.@example.com
+See .john@example.com
+See ..john.doe@example.com
+
+#Quoted local-part, so not linked (in the base algorithm)
+See "john\ doe"@example.com
```

Comment:
In your latest spec draft, you write “Implementations are free to extend this algorithm to support such quoted email local-part formats.”

If an implementation does that, then it will fail this test case.
I suggest we remove it.